exec/exec.go

```diff
@@ -168,6 +175,24 @@ func typeOf(tf *selected.TypenameField, resolver reflect.Value) string {
 	return ""
 }
 
+// SelectedFieldFromContext exposes the fields selected in the GraphQL request using the public-facing types.
+// SelectedField struct
+func SelectedFieldFromContext(ctx context.Context) *types.SelectedField {
```

Comment:
im not convinced this should be here

the structure of the changes feels weird

Reply:
as explained in the DMs - the purpose of the method is not exporting the context key from the package
added some additional comment to make it clear